src/SWP/TemplateEngineBundle/Controller/ContainerController.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\TemplateEngineBundle\Controller;
+
+use FOS\RestBundle\Controller\FOSRestController;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use SWP\ContentBundle\Factory\KnpPaginatorRepresentationFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use SWP\TemplateEngineBundle\Form\Type\ContainerType;
+
+class ContainerController extends FOSRestController
+{
+    /**
+     * Lists all registered containers
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists all registered containers",
+     *     statusCodes={
+     *         200="Returned on success."
+     *     }
+     * )
+     * @Route("/api/{version}/templates/containers/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_list")
+     * @Method("GET")
+     * @Rest\View(statusCode=200)
+     */
+    public function listAction(Request $request)
+    {
+        $entityManager = $this->get('doctrine')->getManager();
+        $paginator = $this->get('knp_paginator');
+        $containers = $paginator->paginate($entityManager->getRepository('SWP\TemplateEngineBundle\Model\Container')->getAll());
```

Comment:
you can register this repository as a service, the code will be more flexible, easier to test (you wont need to mock nested objects, entityManager to get the repo) and it will not break the LoD rule. I would also create some kind of the interfaces for the Doctrine orm repositories as well as odm repositories which will define some basic methods like persist etc. so we can rely on the interface and not on the concrete implementation of EntityRepository object (I think @m038 has been imlementing such interfaces afaik) what do you think?